controllers/spacebindingrequest/spacebindingrequest_controller.go

```diff
@@ -92,7 +99,360 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	}
 	logger.Info("spacebindingrequest found", "member cluster", memberClusterWithSpaceBindingRequest.Name)
 
-	// todo implement logic
+	if util.IsBeingDeleted(spaceBindingRequest) {
+		logger.Info("spaceBindingRequest is being deleted")
+		return reconcile.Result{}, r.ensureSpaceBindingDeletion(logger, memberClusterWithSpaceBindingRequest, spaceBindingRequest)
+	}
+	// Add the finalizer if it is not present
+	if err := r.addFinalizer(logger, memberClusterWithSpaceBindingRequest, spaceBindingRequest); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	createdOrUpdated, err := r.ensureSpaceBinding(logger, memberClusterWithSpaceBindingRequest, spaceBindingRequest)
```

Comment:
does it make sense to return the boolean value in this case?
creation and update is one-time operation and we don't have to wait until the SpaceBinding is provisioned. 
It either fails for any reason (which means that it would return an error) or it succeeds, which means that we can mark it as done.

Reply:
Good point, thanks! 

I got rid of the boolean in https://github.com/codeready-toolchain/host-operator/commit/0d077fdc7829953cfc5ce89b87be1a06b3e2c325

It was a left over from copying SpaceRequest controller I guess 🤦‍♂️